src/org/opendatakit/briefcase/util/ServerFetcher.java

```diff
@@ -238,88 +250,175 @@ public boolean downloadFormAndSubmissionFiles(List<FormStatus> formsToTransfer)
+      return t;
+    }
+  }
+
+  private ExecutorService getFetchExecutorService() {
+    int downloadThreads = BriefcasePreferences.getBriefcaseParallelPullsProperty()? MAX_CONNECTIONS_PER_ROUTE : 1;
+    return Executors.newFixedThreadPool(downloadThreads, new DownloadThreadFactory());
+  }
+
   private boolean downloadAllSubmissionsForForm(File formInstancesDir, DatabaseUtils formDatabase, BriefcaseFormDefinition lfd,
-      FormStatus fs) {
+                                                FormStatus fs) {
+    int submissionCount = 1, chunkCount = 1;
     boolean allSuccessful = true;
-
     RemoteFormDefinition fd = (RemoteFormDefinition) fs.getFormDefinition();
+    ExecutorService execSvc = getFetchExecutorService();
+    CompletionService<SubmissionChunk> chunkCompleter = new ExecutorCompletionService(execSvc);
+    CompletionService<String> submissionCompleter = new ExecutorCompletionService(execSvc);
 
-    int count = 1;
-    String baseUrl = serverInfo.getUrl() + "/view/submissionList";
+    String oldWebsafeCursorString, websafeCursorString = "";
 
-    String oldWebsafeCursorString = "not-empty";
-    String websafeCursorString = "";
-    for (; !oldWebsafeCursorString.equals(websafeCursorString);) {
-      if ( isCancelled() ) {
-        fs.setStatusString("aborting fetching submissions...", true);
+    chunkCompleter.submit(new SubmissionChunkDownload(fs, fd.getFormId(), websafeCursorString));
+
+    boolean cursorFinished;
+
+    try {
+      do {
+        if (isCancelled()) {
+          fs.setStatusString("aborting fetching submission chunks...", true);
+          EventBus.publish(new FormStatusEvent(fs));
+          return false;
+        }
+
+        fs.setStatusString("processing chunk " + chunkCount + "...", true);
         EventBus.publish(new FormStatusEvent(fs));
-        return false;
+
+        oldWebsafeCursorString = websafeCursorString; // remember what we had...
+        SubmissionChunk chunk;
+        try {
+          chunk = chunkCompleter.take().get();
+          chunkCount += 1;
+          websafeCursorString = chunk.websafeCursorString;
+          cursorFinished = oldWebsafeCursorString.equals(websafeCursorString);
+        } catch (InterruptedException | ExecutionException e) {
+          return false;
+        }
+
+        if (!cursorFinished) {
+          // submit another chunk request so it's ready by the time we finish processing this chunk
+          chunkCompleter.submit(new SubmissionChunkDownload(fs, fd.getFormId(), websafeCursorString));
+        }
+
+        for (String uri : chunk.uriList) {
+          if (isCancelled()) {
+            fs.setStatusString("aborting requesting submissions...", true);
+            EventBus.publish(new FormStatusEvent(fs));
+            return false;
+          }
+          submissionCompleter.submit(new SubmissionDownload(formInstancesDir, formDatabase, lfd, fs, uri));
+        }
+
+        for (int i = 0; i < chunk.uriList.size(); i++) {
```

Comment:
Yes, I didn't use the for-each here because the second loop doesn't actually use the items in the list. It only needs to match up with the first loop, since each submitted job in the first loop should have a matching call to `take()` in the second. I can add a comment to clarify and prevent the reader from having to dive directly into the loop semantics.